tendrl/gluster_integration/objects/volume/atoms/create/__init__.py

```diff
@@ -41,6 +41,43 @@ def run(self):
                 "force": self.parameters.get('Volume.force')
             })
 
+        # mark the bricks locked by the task_id so that they cannot be
+        # picked by other create volume tasks
+        for sub_vol in self.parameters.get('Volume.bricks'):
+            for brick in sub_vol:
+                ip = brick.keys()[0]
+                brick_path = brick.values()[0]
+                try:
+                    # Find node id using ip
+                    node_id = NS._int.client.read("indexes/ip/%s" % ip).value
+                except etcd.EtcdKeyNotFound:
+                    # Find node id using hostname
+                    nodes = NS._int.client.read("nodes/")
+                    for node in nodes.leaves:
+                        hostname = NS._int.client.read(
+                            "%s/NodeContext/fqdn" % node.key).value
+                        if hostname == ip:
+                            node_id = node.key.split("/")[-1]
+                key = "nodes/%s/NodeContext/fqdn" % node_id
+                host = NS._int.client.read(key).value
```

Comment:
you can move line 61 and 62 into try
key = "nodes/%s/NodeContext/fqdn" % node_id
host = NS._int.client.read(key).value

because here you are identifying hostname, but in exception block already host name is there (ip is hostname). 
and you can assign host = ip in exception block is enaf